src/proxy/outbound.rs

```diff
@@ -357,16 +434,69 @@ impl OutboundConnection {
             }
             debug!("built request as passthrough; no upstream found");
             return Ok(Request {
-                protocol: Protocol::TCP,
+                protocol: OutboundProtocol::TCP,
                 source: source_workload,
                 hbone_target_destination: None,
                 actual_destination_workload: None,
                 intended_destination_service: None,
                 actual_destination: target,
                 upstream_sans: vec![],
+                final_sans: vec![],
             });
         };
 
+        // Check whether we are using an E/W gateway and sending cross network traffic
+        if let Some(ew_gtw) = &us.workload.network_gateway {
+            if us.workload.network != source_workload.network {
+                let svc = us
+                    .destination_service
+                    .as_ref()
+                    .expect("Workloads with network gateways must be service addressed.");
+
+                let (actual_destination, upstream_sans, final_sans) = match &ew_gtw.destination {
```

Comment:
nit: can we just put this all in state.rs like fetch_waypoint but fetch_network_gateway?